File: Aplikacija/EstateExplorer/EstateExplorer/Models/Zgrada.cs
using Microsoft.AspNetCore.Identity;
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System;
using System.Collections.Generic;

namespace EstateExplorer.Models
{
    public class Zgrada
    {
        [Key]
        public int id { get; set; }

        public string Naziv { get; set; } = default!;
        public string Ulica { get; set; } = default!;
        public string BrojZgrade { get; set; } = default!;

        public string BrojKatastarskeParcele { get; set; } = default!;
        public bool Lift { get; set; }

        public int BrojSpratova { get; set; }
        public string Grejanje { get; set; } = default!;
        public byte[]? Image { set; get; }

        public string? Opis { get; set; }
        public string KatastarskaOpstina { get; set; } = default!;
        public List<Nekretnina>? Nekretnine { get; set; }



    }
}

File: Aplikacija/EstateExplorer/EstateExplorer/ClientApp/src/models/Zgrada.js
class Zgrada {
    id;
    naziv;
    ulica;
    brojZgrade;
    brojKatastarskeParcele;
    lift;
    brojSpratova;
    grejanje;
    opis;
    katastarskaOpstina;
    stanovi;

    constructor(zgrada = undefined) {
        if(zgrada === undefined) {
            this.id = -1;
            this.naziv = 'naziv zgrade';
            this.ulica = 'ulica';
            this.brojZgrade = 'broj zgrade';
            this.brojKatastarskeParcele = 'broj KP';
            this.lift = true;
            this.brojSpratova = 1;
            this.grejanje = 'Tip grejanja';
            this.opis = 'opis zgrade';
            this.katastarskaOpstina = 'KO';
            this.stanovi = [];
        }
        else {
            this.id                     = zgrada.id                     ;
            this.naziv                  = zgrada.naziv                  ;
            this.ulica                  = zgrada.ulica                  ;
            this.brojZgrade             = zgrada.brojZgrade             ;
            this.brojKatastarskeParcele = zgrada.brojKatastarskeParcele ;
            this.lift                   = zgrada.lift                   ;
            this.brojSpratova           = zgrada.brojSpratova           ;
            this.grejanje               = zgrada.grejanje               ;
            this.opis                   = zgrada.opis                   ;
            this.katastarskaOpstina     = zgrada.katastarskaOpstina     ;
            this.stanovi                = zgrada.stanovi !== undefined ? [...zgrada.stanovi] : undefined;  
        }
    }
}

export default Zgrada;